File: BE/src/main/java/com/example/hotel/model/Booking.java
package com.example.hotel.model;

import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Booking {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;

    private Long orderId;

    private LocalDate checkInDate;

    private LocalDate checkOutDate;

    private Long roomTypeId;

    private String status;

    private Long amount;

    @CreationTimestamp
    private LocalDateTime createdTime;

    @UpdateTimestamp
    private LocalDateTime updatedTime;

}

File: FE-ADMIN/demo/src/app/core/model/booking.ts
export class Booking {
  id: number;
  orderId: number;
  roomTypeId: number;
  roomId: number;
  amount: number
  checkInDate: string;
  checkOutDate: string;

  constructor() {
    this.id = null
    this.orderId = null;
    this.roomTypeId = null;
    this.roomId=  null;
    this.amount= null;
    this.checkInDate=  "";
    this.checkOutDate = "";
  }

}
